src/huggingface_hub/inference/_providers/_common.py

```diff
@@ -116,7 +126,9 @@ def _prepare_mapped_model(self, model: Optional[str]) -> str:
 
         # hardcoded mapping for local testing
         if HARDCODED_MODEL_ID_MAPPING.get(self.provider, {}).get(model):
-            return HARDCODED_MODEL_ID_MAPPING[self.provider][model]
+            return ProviderMappingInfo(
+                provider_id=HARDCODED_MODEL_ID_MAPPING[self.provider][model],
```

Comment:
Let switch from `HARDCODED_MODEL_ID_MAPPING` to `HARDCODED_MODEL_INFERENCE_MAPPING` which would become a `Dict[str, Dict[str, ProviderMappingInfo]]`